src/main/java/hudson/plugins/ec2/util/ResettableCountDownLatch.java

```diff
@@ -0,0 +1,43 @@
+package hudson.plugins.ec2.util;
+
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicReference;
+
+public class ResettableCountDownLatch {
```

Comment:
Suggestion: add `@Restricted(NoExternalUse.class)` as this isn't really meant to be available to other plugins to use.